chalk-rust-ir/src/lib.rs

```diff
@@ -130,8 +138,11 @@ pub struct TraitDatumBound<I: Interner> {
 pub struct TraitFlags {
     pub auto: bool,
     pub marker: bool,
+    /// JRL What does upstream trait mean?
     pub upstream: bool,
+    /// JRL What fundamental traits are there? Does this just mean #[fundamental] like `Box`?
     pub fundamental: bool,
+    /// JRL Also what are these?
     pub non_enumerable: bool,
```

Comment:
`non_enumerable` just means we can't list all impls for a given trait

Reply:
Does this just mean in practice that

if defined in library and exported then non_enumerable = true
else non_enumerable = false

Reply:
In practice at least rust-analyzer makes every trait `non_enumerable`, because we don't want Chalk to go off trying to find e.g. every type implementing `Iterator` :grimacing:

Reply:
I.e. this isn't about orphan rules or something like that, it's a hint that Chalk shouldn't try enumerating types that implement the trait because there are too many, basically.

Reply:
rustc makes every trait non-enumerable as well. eventually it may just be the only option I think.